### PR TITLE
Add contenttype and contenttypemod Registries

### LINKED ISSUE #107: Add Registry - Content Type
https://www.isdcf.com/site/dcnc-appendix-3-content-type/

Content Type Table (1) on page, include all ContentKinds (with scopes) from:
- 429-7
- 429-16
- 2067-3 (that are not already included in 429-7 or 492-16)

Content Type Modifier Table (2), include corresponding XML elements where applicable.

### PR DESCRIPTION
**Pull request by SteveLLamb**
Closes #107 

A few notes on these registries:
- For contenttypemod the values for code and value in square brackets are meant to show it can variable and will be described as such on the web page. 
- For contenttypemod the metaType is enum values
- Regs are canonicalized, but set to manual order for initial order on page rendering (to replicate current page order)

### CODE REVIEW

File: test/contenttypes.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          "definingDoc": "SMPTE ST 429-7"
+        },
+        "description": "A theatrical feature."
+      }
+    ]))
+  })
+
+  it("invalid", () => {
+    assert.throw(() => validate([
+      "foo"
```
Comment (palemieux) [inline test/contenttypes.js:34]: Maybe a better effort? :)

File: test/contenttypemods.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          "definingDoc": "SMPTE ST 429-10"
+        },
+        "description": "If the product is 3D."
+      }
+    ]))
+  })
+
+  it("invalid", () => {
+    assert.throw(() => validate([
+      "foo"
```
Comment (palemieux) [inline test/contenttypemods.js:37]: Recommend a more realistic bad example

File: src/main/data/contenttypes.json [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    "contentKind": {
+      "definingDoc": "SMPTE ST 429-7",
+      "scope": "http://www.smpte-ra.org/schemas/429-7/2006/CPL#standard-content",
+      "value": "feature"
+    },
+    "description": "A theatrical feature."
+  },
+  {
+    "code": "TLR",
+    "contentKind": {
```
Comment (palemieux) [inline src/main/data/contenttypes.json:13]: _contentKind_ should be _cplContentKind_

File: src/main/data/contenttypes.json [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    "code": "FTR",
+    "contentKind": {
+      "definingDoc": "SMPTE ST 429-7",
+      "scope": "http://www.smpte-ra.org/schemas/429-7/2006/CPL#standard-content",
+      "value": "feature"
+    },
+    "description": "A theatrical feature."
+  },
+  {
+    "code": "TLR",
```
Comment (palemieux) [inline src/main/data/contenttypes.json:12]: _code_ should _dcncCode_

**Review by palemieux — CHANGES_REQUESTED** [review]
What about `dcnc-content-mods` instead of `contenttypemod` to emphasize it is intended for dcnc only?

File: src/main/data/contenttypes.json (hunk 4 above)
Comment (SteveLLamb) [reply src/main/data/contenttypes.json]: Updated.

File: src/main/data/contenttypes.json (hunk 3 above)
Comment (SteveLLamb) [reply src/main/data/contenttypes.json]: Updated.

File: test/contenttypemods.js (hunk 2 above)
Comment (SteveLLamb) [reply test/contenttypemods.js]: Done, added in missing metadata.

File: test/contenttypes.js (hunk 1 above)
Comment (SteveLLamb) [reply test/contenttypes.js]: Done, added in missing scope.

**Comment by SteveLLamb** [comment]
> What about `dcnc-content-mods` instead of `contenttypemod` to emphasize it is intended for dcnc only?

Updated to `contentmodifiers` as data also includes metadata.

**Review by palemieux — CHANGES_REQUESTED** [review]
Add `dcncSortOrder` integer to preserve historical ordering, and enforce it during validation.

**Comment by SteveLLamb** [comment]
> Add `dcncSortOrder` integer to preserve historical ordering, and enforce it during validation.

Done, added 2 addition checks on Test (missing sort order, and sort order not an integer) and added sorting and duplication conditions on Validate.

**Review by palemieux — APPROVED** [review]

**Pull request merged**